src/WebJobs.Extensions.MobileApps/Config/MobileAppsConfiguration.cs

```diff
@@ -53,8 +62,194 @@ public void Initialize(ExtensionConfigContext context)
                 throw new ArgumentNullException("context");
             }
 
-            context.Config.RegisterBindingExtension(
-                new MobileTableAttributeBindingProvider(context.Config, this, context.Config.NameResolver));
+            INameResolver nameResolver = context.Config.GetService<INameResolver>();
+            IConverterManager converterManager = context.Config.GetService<IConverterManager>();
+
+            BindingFactory factory = new BindingFactory(nameResolver, converterManager);
+
+            IBindingProvider outputProvider = factory.BindToGenericAsyncCollector<MobileTableAttribute, MobileTableContext>(typeof(MobileTableAsyncCollector<>), BindForOutput);
+
+            IBindingProvider clientProvider = factory.BindToExactType<MobileTableAttribute, IMobileServiceClient>(BindForClient);
+
+            IBindingProvider queryProvider = factory.BindToGenericItem<MobileTableAttribute>(BindForQuery);
+            queryProvider = factory.AddTypeFilter(IsQueryType, queryProvider);
+            queryProvider = factory.AddValidator<MobileTableAttribute>(ThrowIfGenericArgumentIsInvalid, queryProvider);
```

Comment:
I think the new rules in the SDK addresses these.

---
In reply to: [63600657](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/97#discussion_r63600657) [](ancestors = 63600657,63583395)